scripts/kind-e2e/lib_subctl_gather_test.sh

```diff
@@ -32,13 +32,27 @@ function test_subctl_gather() {
   validate_resource_files $subm_ns 'deployments' 'Deployment' '-l app=submariner-networkplugin-syncer'
   validate_resource_files $subm_ns 'deployments' 'Deployment' '-l app=submariner-lighthouse-agent'
   validate_resource_files $subm_ns 'deployments' 'Deployment' '-l app=submariner-lighthouse-coredns'
+
+  # Service Discovery
+  validate_resource_files all 'serviceexports.multicluster.x-k8s.io' 'ServiceExport'
+  validate_resource_files all 'serviceimports.multicluster.x-k8s.io' 'ServiceImport'
+  validate_resource_files all 'endpointslices.discovery.k8s.io' 'EndpointSlice'
```

Comment:
We don't export any services for this e2e test so the verification is a no-op but that's OK for now.